File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<TreeNode>();
            List<List<Integer>> res = new LinkedList<List<Integer>>();
            if(root == null) return res;
            
            q.offer(root);
            while(!q.isEmpty()){
                    int size = q.size();
                    List<Integer> level = new LinkedList<Integer>();
                    for(int i=0;i<size;i++){
                            if(q.peek().left !=null) q.offer(q.peek().left);
                            if(q.peek().right !=null) q.offer(q.peek().right);
                            level.add(q.poll().val);
                    }     
                    res.add(level);
            }
            return res;
    }
}

File: 102-binary-tree-level-order-traversal/102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
        void bfs(TreeNode* root, int level, vector<vector<int>> &res){
                if(!root) return;
                if(res.size() == level) res.push_back({});
                res[level].push_back(root->val);
                if(root->left) bfs(root->left, level+1, res);
                if(root->right) bfs(root->right, level+1, res);
        }
    vector<vector<int>>levelOrder(TreeNode* root) {
        vector<vector<int>> res;
            
            bfs(root, 0, res);
            return res;
    }
};